src/RecoveryServices/RecoveryServices.Backup.Autorest/custom/Cmdlets/Policy/Edit-AzrecoveryServicesBackupRetentionPolicyClientObject.ps1

```diff
@@ -0,0 +1,1023 @@
+                            if($policyObject.SchedulePolicy.ScheduleRunFrequency -eq "Daily")
+                            {
+                                $policyObject.RetentionPolicy.YearlySchedule.RetentionScheduleWeekly.DaysOfTheWeek = foreach ($DayOfWeek in $YearlyRetentionDaysOfTheWeek){[Microsoft.Azure.PowerShell.Cmdlets.RecoveryServices.Support.DayOfWeek]::new()|
+                                    Add-Member -MemberType NoteProperty -Name "DayOfWeek" -Value $DayOfWeek -PassThru}
+                                $policyObject.RetentionPolicy.YearlySchedule.RetentionScheduleWeekly.DaysOfTheWeek =$YearlyRetentionDaysOfTheWeek
+                            }
+                            elseif($policyObject.SchedulePolicy.ScheduleRunFrequency -eq "Weekly")
+                            {
+                                $validDaysOfWeek = $policyObject.SchedulePolicy.ScheduleRunDay
+                                $selectedDays = $YearlyRetentionDaysOfTheWeek | Where-Object { $validDaysOfWeek -contains $_ }
+                                if ($selectedDays.Count -ne $YearlyRetentionDaysOfTheWeek.Count)
+                                {
+                                    throw "Only schedule run days are valid for weekly schedule."
+                                }
+                                else
+                                {
+                                    $policyObject.RetentionPolicy.YearlySchedule.RetentionScheduleWeekly.DaysOfTheWeek = $selectedDays
+                                }
+                            }
+                            
+                            $policyObject.RetentionPolicy.YearlySchedule.RetentionScheduleWeekly.WeeksOfTheMonth = foreach ($WeekOfMonth in $YearlyRetentionWeeksOfTheMonth){[Microsoft.Azure.PowerShell.Cmdlets.RecoveryServices.Support.WeekOfMonth]::new()|
+                                    Add-Member -MemberType NoteProperty -Name "WeekOfMonth" -Value $WeekOfMonth -PassThru}
+                            $policyObject.RetentionPolicy.YearlySchedule.RetentionScheduleWeekly.WeeksOfTheMonth =$YearlyRetentionWeeksOfTheMonth
+                        }
+                    }
+              }
+          }
+          
+          if($ModifyDifferentialBackup)
+          {
+               # Mandatory Full backup conditions for testing
+               $policyObject.SubProtectionPolicy[0].SchedulePolicy.ScheduleRunFrequency="Weekly"
+               $policyObject.SubProtectionPolicy[0].SchedulePolicy.ScheduleRunDay=@("Monday","Tuesday")
+               $policyObject.SubProtectionPolicy[0].SchedulePolicy.scheduleRunTime=@("2020-09-30T19:30:00Z")
+               $policyObject.SubProtectionPolicy[0].RetentionPolicy.MonthlySchedule.RetentionScheduleWeekly.DaysOfTheWeek=@("Monday","Tuesday")
+               $policyObject.SubProtectionPolicy[0].RetentionPolicy.YearlySchedule.RetentionScheduleWeekly.DaysOfTheWeek=@("Monday","Tuesday")
```

Comment:
Pls remove the hardcoded setting for schedule policy. do we need this in final command ?